lib/elixir/lib/calendar/time.ex

```diff
@@ -98,37 +98,46 @@ defmodule Time do
       {:error, :invalid_time}
       iex> Time.new(23, 59, 60, 999_999)
       {:error, :invalid_time}
+
+      # Invalid microsecond
       iex> Time.new(23, 59, 59, 1_000_000)
-      {:error, :invalid_time}
+      ** (FunctionClauseError) no function clause matching in Time.new/5
 
-      # Invalid precision
-      Time.new(23, 59, 59, {999_999, 10})
-      {:error, :invalid_time}
+      iex> Time.new(23, 59, 59, {999_999, 10})
+      ** (FunctionClauseError) no function clause matching in Time.new/5
 
   """
   @spec new(
           Calendar.hour(),
           Calendar.minute(),
           Calendar.second(),
-          Calendar.microsecond() | integer,
+          Calendar.microsecond() | 0..999_999,
           Calendar.calendar()
         ) :: {:ok, t} | {:error, atom}
   def new(hour, minute, second, microsecond \\ {0, 0}, calendar \\ Calendar.ISO)
 
-  def new(hour, minute, second, microsecond, calendar) when is_integer(microsecond) do
+  def new(hour, minute, second, microsecond, calendar) when microsecond in 0..999_999 do
     new(hour, minute, second, {microsecond, 6}, calendar)
   end
 
-  def new(hour, minute, second, {microsecond, precision}, calendar)
-      when is_integer(hour) and is_integer(minute) and is_integer(second) and
-             is_integer(microsecond) and is_integer(precision) do
-    case calendar.valid_time?(hour, minute, second, {microsecond, precision}) do
+  def new(
+        hour,
+        minute,
+        second,
+        {microsecond_value, microsecond_precision} = microsecond,
+        calendar
+      )
+      when is_integer(hour) and hour >= 0 and
+             is_integer(minute) and minute >= 0 and
+             is_integer(second) and second >= 0 and
+             microsecond_value in 0..999_999 and microsecond_precision in 0..6 do
```

Comment:
Please revert. All of these limitations apply to the Calendar.ISO, it may not apply to others. Probably best to not do changes to this file in this PR.